### DISCUSSION TITLE
Low Power Warning - Adjust battery % to meet user needs

### DISCUSSION THREAD

**Discussion by c5ryan312** [gopro/labs#838] [Feature Requests]
Hello All,

I would find it quite useful to adjust the Low Power Warning from GoPro Labs features. Maybe others would too. Instead of the three beeps presented as the camera finally loses all power, it would be so useful to adjust this to beep _well before_ the camera loses power, say at a different battery percentage. 

For example, I would like to know well before my GoPro Hero11 shuts  down, when the camera is running low on power. Say I would like to hear the beeps when the camera passes 15% battery life, or 10%, or 5%, depending on how much notice a user needs. This would prepare users for the impending shutdown, so it's not a surprise. Nothing worse than being mid-film and battery quits...if we had a notice before the shutdown, we could take proactive action (swap battery or to new power source). Would really love to see such a feature implemented into GoPro Labs! 

Thanks all for reading! 

Ryan

**Comment by dnewman-gpsw** [comment]
You can run a script to do it, so Labs already supports this.

![image](https://github.com/gopro/labs/assets/10856084/97796f67-80ad-410d-9c23-579f6724f9a7)

![Beeps On Low Battery-4_3_2024, 15_21_04](https://github.com/gopro/labs/assets/10856084/95ac156b-906a-40c1-bb95-2ce76b1628c9)

This will beep (when not recording, otherwise it will mess up your audio) once at 15%, twice at 10% and three times at 5%.  Beeps every 10s or so.

**Reply by c5ryan312** [reply]
Thank you for that information, Dave. That's certainly a route I'd like to try, at least for now However, I _am_ looking for the beeps to occur while recording. I mostly use my GoPro for my YouTube fishing channel, I am not overly concerned about losing audio, but I'm terrified of losing video (when the GoPro suddenly ends once power is used up).  My hands are almost always tied and I don't always have view of the GoPro LCD to see battery life. Here is where an audio cue would be immensely useful to me (I'm sure others too). I almost exclusively use QuikCapture to start the camera, and voice control to end recording/power off GoPro. It would be very nice for me to know when I should start worrying about battery life once the camera is rolling. I can't think of another way the GoPro could give me this cue (while recording) without the audio beeps - at least as a Labs option ;)

**Reply by dnewman-gpsw** [reply]
Modify the above to remove the recording option.  You also might want the beeps loud, I set them very low.  

Go to https://gopro.github.io/labs/build/

Click 'Import Image', then load the saved (PNG) of the QR Code above. This way you can edit the script and enhance as needed.

You will also need to make the a BOOT command so that it starts up automatically.

**Comment by c5ryan312** [comment]
Thanks for that, David! You probably already surmised, I am not exactly up
to speed on scripting with GoPros; this is uncharted territory for me. I
desperately need to understand this further, and am going to dive into it
to see if I can make this work. This latest script idea you have seems
perfect for what I'm looking for. And I don't know blockly (yet), but am
willing to play with it a bit to understand it more. The one area that I'm
struggling with is the BOOT command. How is that introduced in blockly? Is
that via a QR command? Say I want to be able to use QuikCapture to initiate
power-up, OR after already having powered up the camera via the side power
button, and then pressing the Record button. I feel I have started down a
rabbit hole...any help would be appreciated:) Thanks again for this very
useful introduction to scripting with GoPros...I can see great
opportunities (but much learning ahead).

Ryan



<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Reply by dnewman-gpsw** [reply]
It is very easy like this:
![image](https://github.com/gopro/labs/assets/10856084/ba3597aa-368c-4eeb-8f4f-696b580fd896)

![Boot with Low Power Beeps-4_4_2024, 08_02_15](https://github.com/gopro/labs/assets/10856084/7062c895-74c8-4650-8c7b-d784bd0ea500)

**Comment by c5ryan312** [comment]
Sincerely appreciated! I see now - I need to work with blockly more. I did
not know you could stack inside the boot with command, like your script.
This is so helpful, and eye opening. Blockly is seeming more and more
intuitive, more so than I previously believed. I've already signed up for a
blockly class online and am going to get started tonight.

Thanks for all the help, David! This has been instrumental in my learning
here.



<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by c5ryan312** [comment]
Hello David! Interestingly, the script does works for me, but only seems to
apply when I power on the GoPro with the power button, then start recording
once the camera is powered on in that fashion. If I use QuikCapture, the
script doesn't seem to function. Is there a secret to having this script
work when using QuikCapture?




<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Reply by dnewman-gpsw** [reply]
I knew some feature are disabled with quick capture, it seem boot commands are too.  So there is currently no workaround for this.

**Comment by c5ryan312** [comment]
Hey, not a problem, David. Especially with my line of filming, I can just
use the side power button - and if I do end up using QuikCapture, at least
I know it's no longer a feature to rely on. Still, it works, and will serve
me well. Once again, thank you for all the help :)




<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Reply by PkumarGithu** [reply]
Hey David,
thank you so much for this script this is really a big help to record my cricket game sessions.
would please be able to guide me to make a louder beep and 20 sec gap between them, please.
thanks in advance.

**Reply by dnewman-gpsw** [reply]
Try first.  Adding a 20sec delay is straight forward, but the script above using oV9, the loudest beep offered.

Command language documents: https://gopro.github.io/labs/control/tech/

**Reply by PkumarGithu** [reply]
got it, thanks, much appreciated
